### PR TITLE
Annotate type of Plugin workflow argument

### PR DESCRIPTION
**Pull request by tkdchen**
By adding this type annotation, it will make code edit tools, like IDE
or LSP backend, happy and be easy to navigate to the related
symbols.

Signed-off-by: Chenxiong Qi <cqi@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] Python type annotations added to new code
- [n/a] JSON/YAML configuration changes are updated in the relevant schema
- [n/a] Changes to metadata also update the documentation for the metadata
- [n/a] Pull request has a link to an osbs-docs PR for user documentation updates
- [n/a] New feature can be disabled from a configuration file

### CODE REVIEW

**Review by MartinBasti — APPROVED** [review]

File: atomic_reactor/inner.py [hunk 1]
```diff
@@ -668,12 +659,22 @@ def build_process_failed(self):
...
         build docker image
         """
+        from atomic_reactor.plugin import (
+            BuildStepPluginsRunner,
+            ExitPluginsRunner,
+            PluginFailedException,
+            PostBuildPluginsRunner,
+            PreBuildPluginsRunner,
+            PrePublishPluginsRunner,
+        )
```
Comment (chmeliik) [inline atomic_reactor/inner.py:676]: Is this the only way to get around the cyclic import? It feels strange to do it like this :(

**Review by ben-alkov — APPROVED** [review]

File: atomic_reactor/inner.py (hunk 1 above)
Comment (tkdchen) [reply atomic_reactor/inner.py]: PTAL

File: atomic_reactor/plugins/pre_bump_release.py [hunk 2]
```diff
@@ -77,7 +77,7 @@ def next_release_general(
         """Get next release for build."""
         if is_scratch_build(self.workflow):
             # no need to append for scratch build
-            next_release = self.workflow.user_params.get('pipeline_run_name')
+            next_release = self.workflow.user_params['pipeline_run_name']
```
Comment (lkolacek) [inline atomic_reactor/plugins/pre_bump_release.py:80]: Why this change? to make it fail with `KeyError` when `pipeline_run_name` is not available?

Comment (chmeliik) [reply atomic_reactor/plugins/pre_bump_release.py:80]: IIUC: the code doesn't work if `next_release` is None, this change ensures that it's a string (or KeyError, yeah)

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

Nice find on `typing.TYPE_CHECKING` :rocket:

**Review by hjmodi — APPROVED** [review]

**Review by lkolacek — APPROVED** [review]
